task-maker-async/src/client.rs

```diff
@@ -430,6 +433,233 @@ fn prepare_execution_group(
     Some(ret)
 }
 
+// TODO(veluca): consider using the tokio versions of file io.
+async fn read_file_to_memory(
+    store: &StoreClient,
+    file: &FileHandle,
+    size_limit: Option<usize>,
+) -> Result<Vec<u8>, Error> {
+    let mut result = vec![];
+    loop {
+        if let Some(limit) = size_limit {
+            if result.len() >= limit {
+                break;
+            }
+        }
+        let chunk = store
+            .read_chunk(context::current(), *file, result.len())
+            .await??;
+        match chunk {
+            FileReadingOutcome::Dropped => {
+                result.clear();
+            }
+            FileReadingOutcome::EndOfFile => {
+                break;
+            }
+            FileReadingOutcome::Data(chunk) => {
+                result.extend(chunk);
+            }
+        };
+    }
+    Ok(result)
+}
+
+async fn read_file_to_disk(
+    store: &StoreClient,
+    file: &FileHandle,
+    destination: &Path,
+    make_executable: bool,
+) -> Result<(), Error> {
+    std::fs::create_dir_all(destination.parent().unwrap())?;
+    let mut destination = std::fs::File::create(destination)?;
```

Comment:
Are the functions called on `destination` blocking?

Reply:
Yup - that's why there's a "consider tokio io" TODO somewhere

Reply:
Done ;)